openair2/RIC_AGENT/E2AP/e2ap_generate_messages.c

```diff
@@ -178,6 +181,115 @@ int e2ap_generate_e2_setup_request(ranid_t  ranid,
+                   uint8_t **buffer,uint32_t *len,
+                   e2node_type_t e2node_type)
+{
+  E2AP_E2AP_PDU_t pdu;
+  E2AP_E2nodeConfigurationUpdate_t *req;
+  E2AP_E2nodeConfigurationUpdate_IEs_t *ie;
+  E2AP_E2nodeComponentConfigUpdate_ItemIEs_t *updateItemIe;
+  E2AP_E2nodeComponentConfigUpdate_Item_t *updateItem;
+  //E2AP_RANfunction_ItemIEs_t *ran_function_item_ie;
+  //E2AP_E2nodeComponentConfigUpdate_ItemIEs_t *e2node_comp_cfg_update_ie;
+  //ric_ran_function_t *func;
+  //unsigned int i;
+
+  //DevAssert(ric != NULL);
+  DevAssert(buffer != NULL && len != NULL);
+
+  memset(&pdu,0,sizeof(pdu));
+  pdu.present = E2AP_E2AP_PDU_PR_initiatingMessage;
+  pdu.choice.initiatingMessage.procedureCode = E2AP_ProcedureCode_id_E2nodeConfigurationUpdate;
+  pdu.choice.initiatingMessage.criticality = E2AP_Criticality_reject;
+  pdu.choice.initiatingMessage.value.present = E2AP_InitiatingMessage__value_PR_E2nodeConfigurationUpdate;
+  req = &pdu.choice.initiatingMessage.value.choice.E2nodeConfigurationUpdate;
+
+  /* Transaction Id */
+  ie = (E2AP_E2nodeConfigurationUpdate_IEs_t *)calloc(1,sizeof(*ie));
+  ie->id = E2AP_ProtocolIE_ID_id_TransactionID;
+  ie->criticality = E2AP_Criticality_reject;
+  ie->value.present = E2AP_E2nodeConfigurationUpdate_IEs__value_PR_TransactionID;
+  ie->value.choice.TransactionID = 1;
+
+  ASN_SEQUENCE_ADD(&req->protocolIEs.list,ie);
+
+  /* GlobalE2node_ID */
+  ie = (E2AP_E2nodeConfigurationUpdate_IEs_t *)calloc(1,sizeof(*ie));
+  ie->id = E2AP_ProtocolIE_ID_id_GlobalE2node_ID;
+  ie->criticality = E2AP_Criticality_reject;
+  ie->value.present = E2AP_E2nodeConfigurationUpdate_IEs__value_PR_GlobalE2node_ID;
+  global_e2_node_id(ranid, &ie->value.choice.GlobalE2node_ID);
+
+  ASN_SEQUENCE_ADD(&req->protocolIEs.list,ie);
+
+  /* Component Configuration Update List */
+  ie = (E2AP_E2nodeConfigurationUpdate_IEs_t *)calloc(1,sizeof(*ie));
+  ie->id = E2AP_ProtocolIE_ID_id_E2nodeComponentConfigUpdate;
+  ie->criticality = E2AP_Criticality_reject;
+  ie->value.present = E2AP_E2nodeConfigurationUpdate_IEs__value_PR_E2nodeComponentConfigUpdate_List;
+
+  
+  if (e2node_type == E2NODE_TYPE_ENB_CU)
+  { 
+    /* S1 Interface Component Config */
+    updateItemIe = (E2AP_E2nodeComponentConfigUpdate_ItemIEs_t *)calloc(1,sizeof(*updateItemIe));
+    updateItemIe->id = E2AP_ProtocolIE_ID_id_E2nodeComponentConfigUpdate_Item;
+    updateItemIe->criticality = E2AP_Criticality_reject;
+    updateItemIe->value.present = E2AP_E2nodeComponentConfigUpdate_ItemIEs__value_PR_E2nodeComponentConfigUpdate_Item;
+
+    updateItem = (E2AP_E2nodeComponentConfigUpdate_Item_t *)&updateItemIe->value.choice.E2nodeComponentConfigUpdate_Item;
+    updateItem->e2nodeComponentInterfaceType = E2AP_E2nodeComponentInterfaceType_s1;
+    updateItem->e2nodeComponentID.present = E2AP_E2nodeComponentID_PR_e2nodeComponentInterfaceTypeS1;
+    updateItem->e2nodeComponentID.choice.e2nodeComponentInterfaceTypeS1.mme_name.buf = (uint8_t *)strdup("MME1");
```

Comment:
Similar to the CU-UP ID and DU-ID, for the future, it would be good if we can remove and parameterize this hardcoded mme_name.

Reply:
Of course, at this moment, it should be fine.

Reply:
ok